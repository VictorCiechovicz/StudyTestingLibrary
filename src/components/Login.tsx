'use client'

import React, { useState, CSSProperties } from 'react'

const Login = () => {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault()

    console.log(`Usuário: ${username}, Senha: ${password}`)
  }

  const styles: { [key: string]: CSSProperties } = {
    form: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
     
    },
    label: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center'
    },
    input: {
      width: 200,
      height: 40  ,
      textAlign: 'start',
      margin: '10px 0',
      padding: 5
    },
    button: {
      width: 200,
      height: 50,
      borderRadius: 10,
      backgroundColor: '#5bcccb',
      marginTop:10
    }
  }
  return (
    <form onSubmit={handleSubmit} style={styles.form}>
      <label style={styles.label}>
        Usuário
        <input
          type="text"
          value={username}
          onChange={e => setUsername(e.target.value)}
          data-testid="username-input"
          style={styles.input}
        />
      </label>
      <label style={styles.label}>
        Senha
        <input
          type="password"
          value={password}
          onChange={e => setPassword(e.target.value)}
          data-testid="password-input"
          style={styles.input}
        />
      </label>
      <button style={styles.button} type="submit" data-testid="submit-button">
        Login
      </button>
    </form>
  )
}

export default Login
