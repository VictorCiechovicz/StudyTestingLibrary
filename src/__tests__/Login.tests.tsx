// __tests__/Login.test.tsx
import { render, fireEvent, screen } from '@testing-library/react';
import Login from '../components/Login';

test('permite ao usuÃ¡rio fazer login', () => {
  render(<Login />);
  
  fireEvent.change(screen.getByTestId('username-input'), {
    target: { value: 'meuUsuario' },
  });
  
  fireEvent.change(screen.getByTestId('password-input'), {
    target: { value: 'minhaSenha' },
  });

  fireEvent.click(screen.getByTestId('submit-button'));


});
