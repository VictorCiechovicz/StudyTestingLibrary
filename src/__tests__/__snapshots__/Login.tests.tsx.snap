// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login form correctly 1`] = `
<DocumentFragment>
  <form>
    <label>
      Usu√°rio:
      <input
        data-testid="username-input"
        type="text"
        value=""
      />
    </label>
    <label>
      Senha:
      <input
        data-testid="password-input"
        type="password"
        value=""
      />
    </label>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
